runtime/oti/jvmtiInternal.h

```diff
@@ -560,8 +564,19 @@ typedef struct jvmtiGcp_translation {
 	((J9CLASS_DEPTH(clazz) == 0) ? NULL : \
 		(clazz)->superclasses[J9CLASS_DEPTH(clazz) - 1])
 
-#define THREAD_DATA_FOR_VMTHREAD(j9env, vmThread) \
-	((J9JVMTIThreadData *) omrthread_tls_get((vmThread)->osThread, (j9env)->tlsKey))
+/* These macros get and set the J9JVMTIThreadData struct pointer value in a TLS array element. */
+#if JAVA_SPEC_VERSION >= 19
+#define GET_THREAD_DATA_FOR_THREAD(j9env, currentThread, thread) \
+	(jvmtiTLSGet((currentThread)->javaVM, (thread), (j9env)->tlsKey))
+#define SET_THREAD_DATA_FOR_THREAD(j9env, currentThread, thread, value) \
+	(jvmtiTLSSet((currentThread)->javaVM, (thread), (j9env)->tlsKey, (value)))
```

Comment:
Why do we need two variants of these? I thought `jvmtiTLSGet|Set` would wrap the call to the old TLS implementation.

Reply:
I wanted the jvmtiTLS functions to have consistent signatures (passing vm as first parameter). I could change it to take `currentOrVMThread` as the first parameter instead.